### PR TITLE
Fix challenges for H2 and H2016

### LINKED ISSUE #93: Different challenges for different games
- Original thread: https://discord.com/channels/826809653181808651/1067796022199332890
- Originally tracked in #90. It now has its own issue because it was not introduced by v6 since it already existed in v5, and getting it fixed before v6 releases might not be a priority. 
- This bug arises from challenges belonging in different categories in different game versions.
  - E.g. in H1, "The Classics" challenges like silent assassin are in the "assassination" category. 
  - For each location, we might need a separate "challenges.json" file for each game version in the future. 
- [x] Add challenges data in `json` files for all three games.
- [x] Support looking up challenges using game version.
- [x] Support difficulty-aware challenges in H1 & H2.

### PR DESCRIPTION
**Pull request by moonysolari**
## Features
- H2 and H2016 now have their own challenge `json`s, such that a location can have up to three different challenge `json` files.
- Challenges are further organized based on game version.
- Added 4 global "The Classics" challenges for H2, all of which work and can be completed.
- Difficulty-aware challenges in H2 are supported.
- `extractChallengeData.mjs` now supports getting challenge data for different game versions.

## Fixes
- Fixes the disconnection when attempting to open the CAREER -> CHALLENGES page on H1. 
- Resolves #93 .
- Added missing global challenges for H3.

## Caveats
- H1 "pro1" difficulty should use a separate set of challenges. They are not added by this PR.
- H1 currently does not support location-wide challenges because it lacks the `Type` field in its challenge definitions.
- In general, the challenges work, but not every challenge was manually tested to work.

### CODE REVIEW

File: contractdata/FACILITY/_FACILITY_CHALLENGES.json [hunk 1]
```diff
@@ -28,6 +29,7 @@
                     "ParentLocationId": "LOCATION_PARENT_ICA_FACILITY",
                     "Type": "contract",
                     "DifficultyLevels": [],
+                    "OrderIndex": 10000,
```
Comment (AnthonyFuller) [inline contractdata/FACILITY/_FACILITY_CHALLENGES.json:32]: This goes for all challenge JSON files now with this property:
Do we need to do anything with this? Or will the game sort it out for us?

File: contractdata/PARIS/oldparischal.json-bak [hunk 2]
```diff
@@ -0,0 +1,5449 @@
+{
```
Comment (AnthonyFuller) [inline contractdata/PARIS/oldparischal.json-bak:1]: What is this file?

File: components/menuData.ts [hunk 3]
```diff
@@ -628,6 +628,7 @@ menuDataRouter.get(
                     controller.challengeService.getChallengesForContract(
                         getSession(req.jwt.unique_name).contractId,
                         req.gameVersion,
+                        // TODO: Should a difficulty be passed here?
```
Comment (AnthonyFuller) [inline components/menuData.ts:631]: Probably? The game might handle it though (I doubt this).

**Review by AnthonyFuller — CHANGES_REQUESTED** [review]
LGTM, respond to comments.

File: contractdata/FACILITY/_FACILITY_CHALLENGES.json (hunk 1 above)
Comment (moonysolari) [reply contractdata/FACILITY/_FACILITY_CHALLENGES.json:32]: I guess it has to do with the ordering of the challenges and the game probably won't sort it out automatically.

File: contractdata/PARIS/oldparischal.json-bak (hunk 2 above)
Comment (moonysolari) [reply contractdata/PARIS/oldparischal.json-bak]: Thanks for spotting this. This is a backup file of the original `_PARIS_CHALLENGES.json` that I forgot to delete.

File: components/menuData.ts (hunk 3 above)
Comment (moonysolari) [reply components/menuData.ts:631]: What is `/missionrewards` used for?

**Comment by moonysolari** [comment]
Made global challenges version-aware and added missing global challenges. Fixed merge conflicts and now ready for merging.

**Review by RDIL — APPROVED** [review]
RSLGTM

**Review by AnthonyFuller — APPROVED** [review]
LGTM

**Review by AnthonyFuller — CHANGES_REQUESTED** [review]
Due to #174 being merged, all challenge JSONs in this PR must be revised and updated to the "new" and "proper" format (aka the intended format).

Should probably rebase and fix the merge conflict.

**Review by AnthonyFuller — APPROVED** [review]
LGTM.

**Pull request merged**